src/wallet/types.ts

```diff
@@ -373,6 +373,56 @@ export interface FullNodeVersionData {
   maxNumberOutputs: number;
 };
 
+export interface PushNotificationResponseData {
+  success: boolean,
+  error?: string,
+  /** This property shows up in case of validation error. */
+  details?: {message: string, path: string}[]
+};
+
+export interface PushNotificationResult {
+  success: boolean,
+}
+
+export interface PushRegisterRequestData {
+  pushProvider: 'ios' | 'android',
+  deviceId: string,
+  enablePush?: boolean,
+  enableShowAmounts?: boolean,
+};
+
+export interface PushRegisterResponseData extends PushNotificationResponseData {};
+
+export interface PushUpdateRequestData {
+  deviceId: string,
+  enablePush: boolean,
```

Comment:
In the Update it's required, and in the Register it's not?

Reply:
Yes. Because in the registration there are default values. In the update, the payload must be complete we should not assume anything. The operation in which the payload can be optional is the `PATCH`.